src/devices/Display/BiColorBarGraph.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.I2c;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// Represents a 24-Segment bargraph that can display multiple colors.
+    /// </summary>
+    // Product: https://www.adafruit.com/product/1721
+    public class BiColorBarGraph : Ht16k33
+    {
+        private byte[] _displayBuffer = new byte[7];
```

Comment:
`private readonly` maybe for both?

Reply:
Both of those are being assigned to in the code. That means they cannot be `readonly`, right?

Reply:
Use `_displayBuffer.AsSpan().Clear();` on L47 then it can be `readonly`, as nothing -- except the ctor -- assigns a new reference to the fields.

Writing to an array element of a readonly array-field doesn't violate "readonly" if that was kind of the question.

Reply:
Got it. Repeated what you proposed on the other PR. Thanks for showing me!